s3_upload/s3_upload.py

```diff
@@ -288,15 +292,56 @@ def monitor_directories_for_upload(config, dry_run):
             failed_files=failed_upload,
         )
 
+        if log_data["uploaded"]:
+            upload_state = "Fully"
+            successfully_uploaded.append(log_data["run_id"])
+        else:
+            upload_state = "Partially"
+            failed_upload.append(log_data["run_id"])
```

Comment:
Is it fine that the previously empty list variable `failed_upload` that you're appending to here has already been assigned a value from the `multi_core_upload()` function? So we're appending to the output of `multi_core_upload` here?